File: haskell/src/clase1/PatternMatchingAbs.hs
module PatternMatchingAbs where

absInt x = if x < 0 then -x else x

absList (head : tail) = absInt(head) : absList(tail)
absList [] = []

main = do
    print (absList [22, -14, 13, -1, -2, 4])
    print (absList [])

File: scala/src/main/scala/clase1/PatternMatching_Abs.scala
package clase1

object PatternMatching_Abs extends App {

  val absInt = (x: Int) => if (x < 0) -x else x

  def abs(list: List[Int]): List[Int] = list match {
    case head :: tail => absInt(head) :: abs(tail)
    case _ => list
  }

  println(abs(List(22, -14, 13, -1, -2, 4)))
  println(abs(List()))

}